File: Npc/11004058.cs
using Maple2.Model.Enum;
using Maple2.Script.Npc;

/// <summary>
/// 11004058: Orde
/// </summary>
public class _11004058 : NpcScript {
    internal _11004058(INpcScriptContext context) : base(context) {
        Id = 10;
        // TODO: RandomPick 10
    }

    public override bool Next(int selection = 0) {
        switch (Id) {
            case 0:
                // $script:0614185307010089$ 
                // - I'm... sorry about what happened to the leaders of Terrun Calibre.
                return true;
            case 10:
                // $script:0614185307010090$ 
                // - I'm... sorry about what happened to the leaders of Terrun Calibre.
                return true;
            default:
                return true;
        }
    }
}

File: Npc/11004058.py
""" 11004058: Orde """
from npc_api import Option, Script
import random


class Main(Script):
    def first(self) -> int:
        return 10

    def select(self) -> int:
        return 0

    def __0(self, pick: int) -> int:
        # $script:0614185307010089$
        # - I'm... sorry about what happened to the leaders of Terrun Calibre.
        return None # TODO

    def __10(self, pick: int) -> int:
        # $script:0614185307010090$
        # - I'm... sorry about what happened to the leaders of Terrun Calibre.
        return -1

    def button(self) -> Option:
        if (self.state, self.index) == (10, 0):
            return Option.CLOSE
        return Option.NONE
